File: Semestr 1/TablicaChar/C++/Zadanie KOD/Kod_szesnastkowy.cpp
/*autor: Pawel Sadowski*/
/* Program rozwiazujacy zadanie kod ze szkopula
   Konwertuje kod szesnastkowy na kod ASCII i odpowiadajacy znak */

#include <iostream>
#include <cctype>
using namespace std;

int main()
{
    char p[2];
    int d=0;
    int pow=1;

    for(int i=0; i<2; i++)
    {
        cin >> p[i];
        p[i] = toupper(p[i]);
    }


    for(int i=1; i>=0; i--)
    {
        if(p[i] >= '0'&& p[i] <= '9')
        {
            d += (p[i]-'0') * pow;
        }
        else if(p[i] >= 'A' && p[i] <= 'F')
            d += ((p[i]-'A')+10) * pow;
        else
            continue;
        pow  *= 16;
    }
    cout << d << endl;
    cout << char(d);
    return 0;
}

File: Semestr 1/TablicaChar/Python/Zadanie KOD/Kod_szesnastkowy.py
"""
autor: Pawel Sadowski
Program konwertuje kod szesnastkowy na kod ASCII i odpowiadający znak
"""
p = []
for i in range(2):
    znak = input(f"Podaj znak {i+1}: ").strip().upper()
    p.append(znak)

d = 0
pow = 1

for i in range(1, -1, -1):
    if '0' <= p[i] <= '9':
        d += (ord(p[i]) - ord('0')) * pow
    elif 'A' <= p[i] <= 'F':
        d += (ord(p[i]) - ord('A') + 10) * pow
    pow *= 16


print(d)
print(chr(d))
